modxcms/revolution
# Grid filtering refactor for improved accuracy and persistence via URL parameters

### What does it do?
Built upon work merged in #16089 and #16355 to provide better filtering functionality and persistence for the majority of grids. Additional methods were added to the Tabs and Grids base classes to further optimize the code (_e.g._, centralizing the repetitive task of creating the query and clear buttons, which are for the most part exactly the same everywhere they are applied). 

### Why is it needed?
To apply the new global persistence feature to the referenced grids and improve their UIs.

### Grids covered

1. **Media Sources** - Media > Media Sources (/manager/?a=source)
2. **Packages** - Extras > Installer (/manager/?a=workspaces)
3. **Users** - Manage > Users (/manager/?a=security/user)
4. **User Settings** - Manage > Users > [User] Settings (manager/?a=security/user/update&id=1)
5. **Messages** - User (user icon) > Messages (/manager/?a=security/message)
6. **System Settings** - Settings (gear) > System Settings (/manager/?a=system/settings)
7. **System Events** - Settings (gear) > System Settings, 2nd tab (/manager/?a=system/settings)
8. **Form Customization Profiles** - Settings (gear) > Form Customization (/manager/?a=security/forms)
9. **Profile** - Settings (gear) > Form Customization > [Profile] (manager/?a=security/forms/profile/update&id=1)
10. **Dashboards** - Settings (gear) > Dashboards (/manager/?a=system/dashboards)
11. **Widgets** - Settings (gear) > Dashboards, 2nd tab (/manager/?a=system/dashboards)
12. **Contexts** - Settings (gear) > Contexts (/manager/?a=context)
13. **Context Settings** - Settings (gear) > Contexts > [Context], 2nd tab (/manager/?a=context/update&key=[yourcontext])
14. **User Groups & Users** - Settings (gear) > Access Control Lists [click UG in left tree] (/manager/?a=security/permission&group=1)
15. **Access Policies** - Settings (gear) > Access Control Lists, 3rd tab (/manager/?a=security/permission)
16. **Policy Templates** - Settings (gear) > Access Control Lists, 4th tab (/manager/?a=security/permission)
17. **Access Permissions/Contexts** - Settings (gear) > Access Control Lists > [User Group], 2nd tab (manager/?a=security/usergroup/update&id=1)
18. **Access Permissions/Resource Groups** - (same path as # 17, 2nd vertical tab)
19. **Access Permissions/Element Categories** - (same path as # 17, 3rd vertical tab)
20. **Access Permissions/Media Sources** - (same path as # 17, 4th vertical tab)
21. **Access Permissions/Namespaces** - (same path as # 17, 5th vertical tab)
22. **User Group/Users** - Settings (gear) > Access Control Lists > [User Group], 3rd tab (manager/?a=security/usergroup/update&id=1)
23. **User Group/Settings** - Settings (gear) > Access Control Lists > [User Group], 4th tab (manager/?a=security/usergroup/update&id=1)
24. **Lexicon Management** - Settings (gear) > Lexicons (/manager/?a=workspaces/lexicon)
25. **Namespaces** - Settings (gear) > Namespaces (/manager/?a=workspaces/namespace)
26. **Trash Bin** - [Trash icon in Resources/Elements/Categories tab panel] (/manager/?a=resource/trash)
27. **Template/Template Variables** - Elements Tab > [Template], 2nd tab (/manager/?a=element/template/update&id=1)
28. **Template Variables/Template Access** - Elements Tab > [Template Variable], 4th tab (/manager/?a=element/tv/update&id=1)
29. **Plugins/System Events** - Elements Tab > [Plugin], 2nd tab (/manager/?a=element/plugin/update&id=1)

### How to test

1. Ensure all caches a cleared.
2. Visit all grids with filtering capabilities and perform various search queries and clear them, verifying expected behavior.
3. For each grid, perform a search and copy the URL in your browser. Paste that URL in a new window or browser tab to verify navigation to the correct tab, and that the search query is intact.

### Related issue(s)/PR(s)
Replaces the closed PRs: #16081, #16084, #16063, #15946, #15942, and #15935
